### PR TITLE
dev/reference/tools/dm: support new column-mapping syntax

### PR DESCRIPTION
**Pull request by kennytm**
Update documents to include pingcap/tidb-tools#236.

### CODE REVIEW

**Comment by kennytm** [comment]
PTAL @csuzhangxc @CaitinChen

File: dev/reference/tools/data-migration/features/overview.md [hunk 1]
```diff
@@ -403,8 +403,8 @@ Note the following restrictions:
 Configure the following 3–4 arguments in order:
 
 - `instance_id`: the ID of the upstream sharded MySQL or MariaDB instance (0 <= instance ID <= 15)
-- The schema prefix: used to parse the schema name and get the `schema ID`
-- The table prefix: used to parse the table name and get the `table ID`
+- The schema prefix: used to parse the schema name and get the `schema ID`; can be an empty string to indicate the schema name has no pattern and the `schema ID` is always zero
```
Comment (csuzhangxc) [inline dev/reference/tools/data-migration/features/overview.md]: indicate `instance_id` can also be empty?

**Comment by kennytm** [comment]
pingcap/dm#158 is merged.

**Comment by csuzhangxc** [comment]
@lilin90 PTAL

File: dev/reference/tools/data-migration/features/overview.md [hunk 2]
```diff
@@ -386,35 +386,48 @@ column-mappings:
...
     - Each schema of each instance supports 256 tables at most (0 <= table ID <= 255).
     - The ID range of the auto-increment primary key is "0 <= ID <= 17592186044415".
-    - The `{instance ID、schema ID、table ID}` group must be unique.
+    - The `{instance ID, schema ID, table ID}` group must be unique.
 - Currently, the `partition id` expression is a customized feature. If you want to modify this feature, contact the corresponding developers.
 
 **`partition id` arguments configuration**
 
-Configure the following three arguments in order:
+Configure the following 3–4 arguments in order:
```
Comment (yikeke) [inline dev/reference/tools/data-migration/features/overview.md]:
```suggestion
Configure the following three or four arguments in order:
```

File: dev/reference/tools/data-migration/features/overview.md [hunk 3]
```diff
@@ -386,35 +386,48 @@ column-mappings:
...
 
 - `instance_id`: the ID of the upstream sharded MySQL or MariaDB instance (0 <= instance ID <= 15)
-- The schema prefix: used to parse the schema name and get the `schema ID`
-- The table prefix: used to parse the table name and get the `table ID`
+- `schema prefix`: used to parse the schema name and get the `schema ID`
+- `table prefix`: used to parse the table name and get the `table ID`
+- The separator: used to separate between the prefix and the IDs, and can be omitted to use an empty string as separator
+
+Any of `instance_id`, `schema prefix` and `table prefix` can be set to an empty string (`""`) to indicate the corresponding parts will not be encoded into the partition ID.
```
Comment (yikeke) [inline dev/reference/tools/data-migration/features/overview.md]:
```suggestion
Any of `instance_id`, `schema prefix` and `table prefix` can be set to an empty string (`""`) to indicate that the corresponding parts will not be encoded into the partition ID.
```

**Comment by kennytm** [comment]
Done @yikeke

**Pull request merged**